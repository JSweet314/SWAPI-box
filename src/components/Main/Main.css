.main {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

.main__btnGroup {
  display: flex;
  justify-content: space-evenly;
  width: 50%;
}