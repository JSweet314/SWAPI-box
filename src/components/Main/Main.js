import React from 'react';
import './Main.css';
import MainBtnGroup from '../MainBtnGroup/MainBtnGroup';
import CategoryDisplay from '../CategoryDisplay/CategoryDisplay';
import PropTypes from 'prop-types';

const Main = ({ currentCategory, selectCategory, categoryResponse }) => {
  return (
    <main className="main">
      <MainBtnGroup 
        currentCategory={currentCategory}
        selectCategory={selectCategory} />
      <CategoryDisplay 
        categoryResponse={categoryResponse}
        currentCategory={currentCategory} />
    </main>
  );
};

Main.propTypes = {
  currentCategory: PropTypes.string.isRequired,
  selectCategory: PropTypes.func.isRequired,
  categoryResponse: PropTypes.array.isRequired
};

export default Main;