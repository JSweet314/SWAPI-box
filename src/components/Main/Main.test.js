import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

describe('Main', () => {
  /* eslint-disable no-undef*/
  const mockChangeFavCount = jest.fn();
  /* eslint-enable no-undef*/
  
  it('should match a snapshot', () => {
    const wrapper = shallow(
      <Main changeFavCount={mockChangeFavCount}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
