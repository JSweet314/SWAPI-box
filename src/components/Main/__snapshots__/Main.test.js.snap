// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="main__btnGroup"
>
        <Button
                btnText="people"
        />
        <Button
                btnText="planets"
        />
        <Button
                btnText="vehicles"
        />
</div>,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            btnText="people"
/>,
          <Button
            btnText="planets"
/>,
          <Button
            btnText="vehicles"
/>,
        ],
        "className": "main__btnGroup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "btnText": "people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "btnText": "planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "btnText": "vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="main__btnGroup"
>
          <Button
                    btnText="people"
          />
          <Button
                    btnText="planets"
          />
          <Button
                    btnText="vehicles"
          />
</div>,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              btnText="people"
/>,
            <Button
              btnText="planets"
/>,
            <Button
              btnText="vehicles"
/>,
          ],
          "className": "main__btnGroup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "btnText": "people",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "btnText": "planets",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "btnText": "vehicles",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
